src/utils.ts

```diff
@@ -0,0 +1,52 @@
+import {ethers} from 'ethers';
+import {chainIdToRpc} from "./constants";
+import {bridgeAbi} from "../abis/bridgeAbi";
+
+export async function getWalletsForDifferentProviders(privateKey: string, networks: Array<any>) {
+  const wallets = [];
+  for (let i = 0; i < networks.length; i++) {
+    const network = networks[i];
+    const chainId: keyof typeof chainIdToRpc = network.chainId;
+    const rpc = chainIdToRpc[chainId];
+    if (rpc) {
+        const provider = new ethers.JsonRpcProvider(rpc);
+        const wallet = new ethers.Wallet(privateKey, provider); // add error handling for invalid private key
+        wallets.push(wallet);
+    }
+  }
+  return wallets;
+}
+
+export async function deriveWalletsFromMnemonic(mnemonic: string, networks: Array<any>) {
+  const wallets = [];
+  for (let i = 0; i < networks.length; i++) {
+    const network = networks[i];
+    const chainId: keyof typeof chainIdToRpc = network.chainId;
+    const rpc = chainIdToRpc[chainId];
+    if (rpc) {
+        const provider = new ethers.JsonRpcProvider(rpc);
+        const wallet = ethers.Wallet.fromPhrase(mnemonic, provider);
+        wallets.push(wallet);
+    }
+  }
+  return wallets;
+}
+
+export async function sendPauseTransactions(networks: Array<any>, wallets: Array<ethers.Wallet>) {
+    const receipts = [];
+    for (let i = 0; i < networks.length; i++) {
+        const network = networks[i];
+        const wallet = wallets[i];
+        const bridge = await getBridgeContractInstance(network, wallet)
+        const tx = await bridge.adminPauseTransfers();
+        const receipt = await tx.wait();
+        console.log(`Transaction no. ${i+1} completed, bridge on ${network.name} paused`);
+        receipts.push(receipt);
+    }
+    return receipts;
+}
+
+async function getBridgeContractInstance(network: any, wallet: ethers.Wallet) {
+    const bridge = new ethers.Contract(network.bridge, bridgeAbi, wallet);
```

Comment:
Actually we can avoid this importing of hardcoded ABI, if you import Bridge factory from Sygma SDK, check how it's done here:
https://github.com/sygmaprotocol/sygma-sdk/blob/82244855040a7835492263a19493b84e3becce86/packages/sdk/src/chains/EVM/assetTransfer.ts#L144

Reply:
Resolved